pkg/decoder/decoder.go

```diff
@@ -52,25 +74,55 @@ func (d *Decoder) Start() {
 	go d.run()
 }
 
+// Stop stops the Decoder
+func (d *Decoder) Stop() {
+	close(d.InputChan)
+}
+
 // run lets the Decoder handle data coming from the InputChan
 func (d *Decoder) run() {
 	for data := range d.InputChan {
-		d.decodeIncomingData(data.content, data.offset)
+		d.stopCountDown()
+		if ok, offset := d.decodeIncomingData(data.content, data.offset); ok {
+			d.restartCountDown(offset)
+		}
```

Comment:
`ok` doesn't sound like a great name, what happens if it's not ok? `ok` sounds like we were able to decode the data